src/compiler/crystal/interpreter/value.cr

```diff
@@ -67,6 +67,15 @@ struct Crystal::Repl::Value
     end
   end
 
+  def runtime_type : Crystal::Type
+    if type.is_a?(Crystal::UnionType)
+      type_id = @pointer.as(Int32*).value
+      context.type_from_id(type_id)
+    else
+      type
+    end
+  end
```

Comment:
**chore:** This feels like it should be introduced as a separate feature. It's not directly related to `#parse_and_interpret`. The specs should probably work fine without it, but we could also stack PRs.

Reply:
Yup. I aimed to that initially but then I remembered the one repl with prelude restriction and somehow push them together.

I'll send another PR for that.

In it we could use this to show type information as part of the repl: eg `1 :: Int32 <: (String | Int32)` when instructed, but I'm not fully sure about the syntax. Or we can leave that usage for later also.